test/bntest.c

```diff
@@ -329,8 +398,10 @@ static int test_mod(void)
         BN_set_negative(b, rand_neg());
         if (!(TEST_true(BN_mod(c, a, b, ctx))
                 && TEST_true(BN_div(d, e, a, b, ctx))
-                && TEST_true(BN_sub(e, e, c))
-                && TEST_BN_eq_zero(e)))
+                && TEST_BN_eq(e, c)
+                && TEST_true(BN_mul(c, d, b, ctx))
+                && TEST_true(BN_add(d, c, e))
```

Comment:
Just a personal comment, which does not request to change anything:

Somehow my brain here would expect rather

~~~
e <- d * b;
d <- e + c;
~~~

to bind together `BN_div()` and `BN_mod()`.

But of course it is absolutely fine as you wrote it, given we tested `e == c`.

Reply:
It's reusing a variable, I just picked one without regards to a hidden or implied meaning.